libvast/vast/fbs/coder.fbs

```diff
@@ -0,0 +1,40 @@
+include "bitmap.fbs";
+
+namespace vast.fbs.coder.detail;
+
+table Base {
+  values: [ulong] (required);
+}
+
+namespace vast.fbs.coder;
+
+table SingletonCoder {
+  bitmap: Bitmap (required);
+}
+
+table VectorCoder {
+  size: ulong;
+  bitmaps: [Bitmap] (required);
+}
+
+table MultiLevelCoder {
+  base: detail.Base (required);
+  xs: [ulong] (required);
```

Comment:
This should be `[ubyte]` so we don't store excess bytes for every value if `sizeof(coder::value_type) < sizeof(uint64_t)`.

Reply:
You can remove `xs` entirely. It's a `mutable` cache and not related to persistent state.

Reply:
Mind if I write a story for you to write documentation for `coder`, `base`, and `bitmap_index`?